tests/test_pytest_writer.py

```diff
@@ -0,0 +1,86 @@
+"""Tests for behave_writer module."""
+import pytest
+
+from sphinx_gherkindoc import pytest_bdd_writer as writer
+import rst_output as _rst_output
+
+
+def _reformat_keywords(rst_lines):
+    for (no_description_keyword, marker) in (("Background", "-"), ("Examples", "~")):
+        line_to_match = (
+            f":gherkin-{no_description_keyword.lower()}-keyword:"
+            f"`{no_description_keyword}:`"
+        )
+        for i, line in enumerate(rst_lines):
+            if line.startswith(line_to_match):
+                rst_lines[i] = f"{line_to_match}\n"
+            if rst_lines[i - 1] == f"{line_to_match}\n":
+                marker_line = marker * len(line_to_match)
+                rst_lines[i] = f"{marker_line}\n\n"
+    return rst_lines
+
+
+@pytest.fixture()
+def rst_output():
+    for attribute in dir(_rst_output):
+        if not attribute.startswith("_"):
+            setattr(
+                _rst_output,
+                attribute,
+                _reformat_keywords(getattr(_rst_output, attribute)),
+            )
+    return _rst_output
+
+
+def _tagline_into_word_set(tag_line):
+    return {word.strip(" ,") for word in tag_line.split() if word}
+
+
+def check_with_tags(actual, expected):
+    tag_text = "Tagged:"
+
+    # pytest-bdd maintains tag data in sets, so the tag order is never guaranteed,
+    # thus we have to evaluate non-tag lines for direct matching,
+    #  and tag lines for all the same "words"
+
+    # First, compare non-tag lines
+    actual_without_tags = [x for x in actual if tag_text not in x]
```

Comment:
Too bad we're not using more_itertools :-)
https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.partition